packages/credential-provider-ini/src/resolveAssumeRoleCredentials.ts

```diff
@@ -116,13 +117,18 @@ export const resolveAssumeRoleCredentials = async (
       {
         ...options.clientConfig,
         credentialProviderLogger: options.logger,
-        parentClientConfig: options?.parentClientConfig,
+        parentClientConfig: {
+          ...options?.parentClientConfig,
+          // The profile region is the last fallback, and only applies
+          // if the clientConfig.region is not defined by the user
+          // and no contextual outer client configuration region can be found.
+          region: options?.parentClientConfig?.region ?? region,
```

Comment:
Should region specification appear before overriding with parentClientConfig?

Reply:
Yes, I was 50/50 during the design phase, but since you pointed it out I believe it makes more sense with higher priority.